lib/game_logic.rb

```diff
@@ -0,0 +1,57 @@
+# Display Board
+require_relative './helper_module.rb'
+class Board
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @board = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+  end
+
+  def display_board
+    <<-GAME
+    \t -------------
+    \t | #{@board[0]} | #{@board[1]} | #{@board[2]} |
+    \t -------------
+    \t | #{@board[3]} | #{@board[4]} | #{@board[5]} |
+    \t -------------
+    \t | #{@board[6]} | #{@board[7]} | #{@board[8]} |
+    \t -------------
+    GAME
+  end
+
+  # Checking and eturning the winner
+  def win_moves(player)
+    temp = []
+    @board.each_with_index do |val, i|
+      player == @player1 ? (temp << i if val == 'X') : (temp << i if val == 'O')
+    end
```

Comment:
#### Optional
It provides current player moves. Maybe the moves generate a winning combination, maybe not. So name of the method does not fit with its operation. Maybe current_player_moves is more suitable. though too long.